uniffi_aries_vcx/demo/app/src/main/java/org/hyperledger/ariesvcx/MainActivity.kt

```diff
@@ -4,78 +4,78 @@ import android.os.Bundle
-                    walletKey = "8dvfYSt5d1taSd6yJdpjq4emkwsPDDLYxkNFysFD2cZY",
-                    walletKeyDerivation = "RAW",
-                    walletType = null,
-                    storageConfig = null,
-                    storageCredentials = null,
-                    rekey = null,
-                    rekeyDerivationMethod = null
-                )
-                mutableStateOf(walletConfig)
-            }
-            val snackbarHostState = remember { SnackbarHostState() }
-            val scope = rememberCoroutineScope()
-            Scaffold(Modifier.fillMaxSize(), snackbarHost = {
-                SnackbarHost(snackbarHostState) { data ->
-                    Snackbar(
-                        modifier = Modifier
-                            .padding(12.dp)
-                    ) {
-                        Text(data.visuals.message)
-                    }
-                }
-            }) {
-                Column(
-                    modifier = Modifier
-                        .padding(it)
-                        .fillMaxSize(),
-                    horizontalAlignment = Alignment.CenterHorizontally,
-                    verticalArrangement = Arrangement.Center
-                ){
-                    Spacer(modifier = Modifier.height(16.dp))
-                    Button(onClick = {
-                        newIndyProfile(walletConfigState)
-                        scope.launch {
-                            snackbarHostState.showSnackbar("New Indy profile: ${walletConfigState}")
-                        }
-                    }) {
-                        Text("Create new profile")
-                    }
+            DemoTheme {
+                Surface (
+                    modifier = Modifier.fillMaxSize(),
+                    color = MaterialTheme.colorScheme.background
+                ) {
+                    val navController = rememberNavController()
+                    NavigationAppHost(
+                        navController = navController,
+                        setProfileHolder = { setProfileHolder(it) },
+                        connection = connection,
+                        profileHolder = profile
+                    )
                 }
             }
         }
     }
 }
+
+@Composable
+fun NavigationAppHost(
+    navController: NavHostController,
+    setProfileHolder: (ProfileHolder) -> Unit,
+    connection: Connection?,
+    profileHolder: ProfileHolder?
+) {
+    NavHost(navController = navController, startDestination = "home") {
+        composable(Destination.Home.route) {
+            HomeScreen(
+                navController = navController,
+                setProfileHolder = setProfileHolder,
+                profileHolder = profileHolder,
+                connection = connection
+            )
+        }
+
+        composable(Destination.QRScan.route) {
+            ScanScreen(
+                connection = connection,
+                profileHolder = profileHolder
```

Comment:
hmmm i wonder if you could set these as non-nullable, since we know they logically cannot be null... e.g. `connect = connection!!, profileHolder = profileHolder!!`. and then ScanScreen could take non-nullable params